code/src/components/MovieList.js

```diff
@@ -0,0 +1,83 @@
+import React, {useEffect, useState} from "react";
+import styled from "styled-components";
+import Header from "./Header";
+import DisplayMovieList from "./DisplayMovieList";
+
+
+
+
+const MovieWrapper = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+
+`
+
+
+
+
+const MovieList = ( {movieList, setMovieList} ) => {
+    
+    
+    const [star, setStar] = useState(true);
+    const [likes, setLikes] = useState(false);
+    const [favourite, setFavourite] = useState([]);
+
+    
+    useEffect(() => {
+      if (star) {
+        setStar(false)
+      }
+      return setStar(true);
+    },[favourite])
+    
+    const onLikeBtnClick = (e, id, movie) => {
+      
+      
+      e.preventDefault();
+      
+      
+      const matchStarId = movieList.find(item => item.id === id );
+      
+      // To toggle between like and dislike
+      
+      setStar(true) 
```

Comment:
We noticed that when clicking on the star for a movie several times it added the same film to the favourites list several times. Also the toggle function does not appear to work.